dev/bots/test/prepare_package_test.dart

```diff
@@ -120,8 +120,8 @@ void main() {
           '$flutter create --template=app ${createBase}app': null,
           '$flutter create --template=package ${createBase}package': null,
           '$flutter create --template=plugin ${createBase}plugin': null,
-          'git clean -f -X **/.packages': null,
-          'git clean -f -X **/.dart_tool': null,
+          'git clean -f -x **/.packages': null,
+          'git clean -f -x **/.dart_tool': null,
```

Comment:
I checked this locally, it appears that `-x` will not delete directories, even if you specify `-d` in addition, and adding `-X` in again will again delete the cache.

Reply:
`¯\_(ツ)_/¯`

Reply:
Ahh.  Try this instead:

```suggestion
          'git clean -f -x **/.dart_tool/': null,
```

Reply:
Without the trailing "/" on the path, it doesn't look for directories.

Reply:
Just `-x` worked on my machine:
```
$ git --version
git version 2.30.1.766.gb4fecdf3b7-goog
$ git clean -f -x **/.dart_tool             
Removing dev/automated_tests/.dart_tool/
Removing dev/benchmarks/complex_layout/.dart_tool/
...
```
and it did delete those directories.

Reply:
Huh.  Was that on macOS or Linux?

I have `git version 2.30.1.766.gb4fecdf3b7-goog` too.

Reply:
Because I get this.  Could be a platform difference.
```
$ git clean -n -x **/.dart_tool
$ git clean -n -x **/.dart_tool/
Would remove packages/flutter_tools/.dart_tool/
$ 
```

Reply:
ahh sorry, I missed that this comment was about the `**/.dart_tool` line.

Reply:
on linux i can confirm that `git clean -f -x **/.dart_tool` doesn't work, and `git clean -f -x -- **/.dart_tool/` DOES work.

Reply:
updated with the slash at the end